test/common/unit/test_tools.py

```diff
@@ -102,3 +110,68 @@ def test_setup_data_folder():
     assert os.path.isdir(test_folder)
 
     os.rmdir(test_folder)
+
+
+def test_intify_unchanged():
+    test_cases = [
+        0,
+        1.1,
+        True,
+        False,
+        "yo",
+        {},
+        [],
+    ]
+    for x in test_cases:
+        assert intify(x) == x
+
+    res = intify([3, 2.5, "yo"])
+    assert res == [3, 2.5, "yo"]
+    assert type(res[0]) == int
+    assert type(res[1]) == float
+
+    x = [3, 2.5, "yo", [4]]
+    res = intify(x)
+    assert res == x
+    assert type(res[0]) == int
+    assert type(res[1]) == float
+    assert type(res[3][0]) == int
+
+    x = {"a": 1, "b": 2.5, "c": {"d": [8]}}
+    res = intify(x)
+    assert res == x
+    assert type(res["a"] == int)
+    assert type(res["b"] == float)
+    assert type(res["c"]["d"][0] == int)
+
+
+def test_intify_changed():
+    assert intify(1.0) == 1
+    assert intify(-1.0) == -1
+
+    x = [1.0, 1.5, -2.0, True]
+    res = intify(x)
+    assert res == [1.0, 1.5, -2, True]
```

Comment:
Shouldn't this be `[1, 1.5, -2, True]`?

Reply:
Yup, although python doesn't check the types. Changed for consistency in 8909d91744bc8298f91aaca2eee48db72ee5e3d4.